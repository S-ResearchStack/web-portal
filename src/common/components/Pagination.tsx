import React, { FC, useCallback, useMemo } from 'react';

import styled from 'styled-components';

import Button from 'src/common/components/Button';
import Dropdown, { MenuItem, ValueItem } from 'src/common/components/Dropdown';
import PageFirstIcon from 'src/assets/icons/page_first.svg';
import PageLastIcon from 'src/assets/icons/page_last.svg';
import ChevronLeftIcon from 'src/assets/icons/chevron_left_pagination.svg';
import ChevronRightIcon from 'src/assets/icons/chevron_right_pagination.svg';
import { colors, px, typography } from 'src/styles';

export const PAGINATION_LIMITS = [10, 25, 50, 75];

export interface PaginationProps {
  disabled?: boolean;
  offset?: number;
  pageNumber?: number;
  totalCount: number;
  pageSize: number;
  onPageChange: (offset: number, size: number, page: number) => void;
}

const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  ${typography.bodySmallRegular};
  line-height: ${px(28)};
`;

const Info = styled.div`
  ${typography.bodySmallRegular};
  color: ${colors.primary};
  line-height: ${px(28)};
`;

const Controls = styled.div`
  align-items: center;
  justify-content: flex-end;
  display: flex;
  margin-right: ${px(-2)};
`;

const ArrowButton = styled(Button).attrs({
  fill: 'text',
  rate: 'icon',
})`
  margin-left: ${px(16)};
  &:disabled {
    > div {
      background-color: transparent;
      svg {
        fill: ${colors.disabled};
      }
    }
  }
`;

const DropDownContainer = styled.div`
  display: flex;
  align-items: center;
  margin-right: ${px(41)};
`;

const DropDownLabel = styled.span`
  ${typography.bodySmallRegular};
  color: ${colors.primary};
  margin-right: ${px(8)};
`;

const DropDownControl = styled(Dropdown)`
  width: ${px(40)};
  height: ${px(25)};
  border: none;
  border-radius: ${px(4)};
  z-index: 10;
  color: ${colors.textSecondaryGray};

  & > ${ValueItem} {
    position: relative;
    z-index: 100;
    display: flex;
    align-items: center;
    justify-content: center;
    padding-left: ${px(6)};
    padding-right: ${px(6)};
    border-color: ${colors.disabled};
    &:hover {
      cursor: pointer;
    }
    > svg {
      display: none;
    }
  }
` as typeof Dropdown;

const SmallMenuItem = styled(MenuItem)`
  background-color: ${({ selected }) => (selected ? colors.primary10 : colors.backgroundSurface)};
  width: ${px(38)};
  padding: 0;
  overflow-x: hidden;

  div {
    width: ${px(38)};
    text-align: center;
    > svg {
      display: none;
    }
  }
`;

const Pagination: FC<PaginationProps> = ({
  totalCount,
  pageSize,
  onPageChange,
  disabled,
  offset,
  pageNumber = 0,
  ...props
}) => {
  const offsetValue = offset != undefined ? offset : pageNumber * pageSize;
  const isPrevDisabled = offsetValue === 0;
  const isNextDisabled = offsetValue + pageSize >= totalCount;
  const sizes = useMemo(() => PAGINATION_LIMITS.map((i) => ({ label: i.toString(), key: i })), []);

  const handleSizeChanged = useCallback(
    (size: number) => !disabled && onPageChange(offsetValue, size, 0),
    [onPageChange, disabled, offsetValue]
  );

  const handlePageClick = useCallback(
    (newOffset: number, newPage: number) => !disabled && onPageChange(newOffset, pageSize, newPage),
    [pageSize, onPageChange, disabled]
  );

  const goToFirst = useCallback(
    () => handlePageClick(0, 0),
    [handlePageClick]
  );
  const goToLast = useCallback(
    () => handlePageClick(totalCount - pageSize, Math.ceil(totalCount / pageSize) - 1),
    [handlePageClick, totalCount, pageSize]
  );

  const goToPrev = useCallback(
    () => handlePageClick(Math.max(offsetValue - pageSize, 0), pageNumber - 1),
    [handlePageClick, offsetValue, pageSize, pageNumber]
  );
  const goToNext = useCallback(
    () => handlePageClick(Math.min(offsetValue + pageSize, totalCount), pageNumber + 1),
    [handlePageClick, offsetValue, pageSize, totalCount, pageNumber]
  );

  return (
    <Container {...props} data-testid="pagination">
      <Info data-testid="info" {...props}>
        {`${offsetValue + 1}-${Math.min(offsetValue + pageSize, totalCount)} of ${totalCount}`}
      </Info>
      <Controls {...props}>
        <DropDownContainer>
          <DropDownLabel>Rows per page</DropDownLabel>
          <DropDownControl
            items={sizes}
            activeKey={pageSize}
            onChange={handleSizeChanged}
            menuItemComponent={SmallMenuItem}
            backgroundType="light"
            menuItemHeight={25}
            maxVisibleMenuItems={5}
          />
        </DropDownContainer>
        <ArrowButton
          disabled={isPrevDisabled}
          onClick={goToFirst}
          data-testid="go-to-first"
          aria-label="Go to First"
        >
          <PageFirstIcon />
        </ArrowButton>
        <ArrowButton
          disabled={isPrevDisabled}
          onClick={goToPrev}
          data-testid="go-to-previous"
          aria-label="Go to Previous"
        >
          <ChevronLeftIcon />
        </ArrowButton>
        <ArrowButton
          disabled={isNextDisabled}
          onClick={goToNext}
          data-testid="go-to-next"
          aria-label="Go to Next"
        >
          <ChevronRightIcon />
        </ArrowButton>
        <ArrowButton
          disabled={isNextDisabled}
          onClick={goToLast}
          data-testid="go-to-last"
          aria-label="Go to Last"
        >
          <PageLastIcon />
        </ArrowButton>
      </Controls>
    </Container>
  );
};

export default Pagination;
