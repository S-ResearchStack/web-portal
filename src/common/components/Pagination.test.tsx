import React from 'react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import theme from 'src/styles/theme';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from 'styled-components/';
import Pagination from './Pagination';

describe('Pagination', () => {
  const TOTAL = 1000;
  const PAGE_SIZE = 10;
  const CURRENT_OFFSET = 40;

  it('test pagination render', async () => {
    const onPageChange = jest.fn();

    const { baseElement, queryByTestId } = render(
      <ThemeProvider theme={theme}>
        <Pagination
          totalCount={TOTAL}
          pageSize={PAGE_SIZE}
          offset={CURRENT_OFFSET}
          onPageChange={onPageChange}
        />
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    const pagination = queryByTestId('pagination') as Element;
    const prevButton = queryByTestId('go-to-previous') as Element;
    const nextButton = queryByTestId('go-to-next') as Element;
    const firstButton = queryByTestId('go-to-first') as Element;
    const lastButton = queryByTestId('go-to-last') as Element;
    const info = queryByTestId('info') as Element;

    expect(pagination).toBeInTheDocument();
    expect(prevButton).toBeInTheDocument();
    expect(nextButton).toBeInTheDocument();
    expect(firstButton).toBeInTheDocument();
    expect(lastButton).toBeInTheDocument();
    expect(info).toBeInTheDocument();

    expect(prevButton).not.toBeDisabled();
    expect(firstButton).not.toBeDisabled();
    expect(nextButton).not.toBeDisabled();
    expect(lastButton).not.toBeDisabled();
    expect(info).toHaveTextContent(
      `${CURRENT_OFFSET + 1}-${CURRENT_OFFSET + PAGE_SIZE} of ${TOTAL}`
    );

    await userEvent.click(nextButton);
    expect(onPageChange).toHaveBeenLastCalledWith(50, 10, 1);

    await userEvent.click(prevButton);
    expect(onPageChange).toHaveBeenLastCalledWith(30, 10, -1);

    await userEvent.click(firstButton);
    expect(onPageChange).toHaveBeenLastCalledWith(0, 10, 0);

    await userEvent.click(lastButton);
    expect(onPageChange).toHaveBeenLastCalledWith(990, 10, 99);
  });

  it('[NEGATIVE] should render with wrong props', async () => {
    const onPageChange = jest.fn();

    const { baseElement, queryByTestId } = render(
      <ThemeProvider theme={theme}>
        <Pagination totalCount={-1} pageSize={-1} offset={-1} onPageChange={onPageChange} />
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    const pagination = queryByTestId('pagination') as Element;
    const prevButton = queryByTestId('go-to-previous') as Element;
    const nextButton = queryByTestId('go-to-next') as Element;
    const firstButton = queryByTestId('go-to-first') as Element;
    const lastButton = queryByTestId('go-to-last') as Element;

    expect(pagination).toBeInTheDocument();
    expect(prevButton).toBeInTheDocument();
    expect(nextButton).toBeInTheDocument();
    expect(firstButton).toBeInTheDocument();
    expect(lastButton).toBeInTheDocument();
  });
});
