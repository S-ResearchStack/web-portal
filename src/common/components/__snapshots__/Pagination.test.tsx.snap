// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination [NEGATIVE] should render with wrong props 1`] = `
.c15 {
  position: absolute;
  will-change: opacity,transform;
  border-radius: 50%;
  -webkit-animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  z-index: 0;
  background-color: #E3EAFB;
  pointer-events: none;
}

.c12 {
  position: relative;
  z-index: 0;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 100%;
  white-space: nowrap;
  padding: 0;
  -webkit-transition-duration: 0;
  transition-duration: 0;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 1.5rem;
  width: 1.5rem;
  border: none;
  background-color: transparent;
}

.c12:hover {
  cursor: pointer;
}

.c12 > div {
  color: #4475E3;
}

.c12 > div svg {
  fill: #4475E3;
}

.c12:hover:enabled > div {
  color: #2F65E0;
}

.c12:hover:enabled > div svg {
  fill: #2F65E0;
}

.c12:hover:enabled:active > div {
  color: #2057D5;
}

.c12:hover:enabled:active > div svg {
  fill: #2057D5;
}

.c12:disabled > div {
  color: #B3C6F1;
}

.c12:disabled > div svg {
  fill: #B3C6F1;
}

.c12:focus-visible {
  outline: 0.125rem solid #C7D6F7;
}

.c12:focus-visible > div {
  color: #4475E3;
}

.c12:focus-visible > div svg {
  fill: #4475E3;
}

.c12:hover:enabled {
  background-color: #ECF1FC;
}

.c12:hover:enabled:active > div {
  color: #2057D5;
}

.c12:hover:enabled:active > div svg {
  fill: #2057D5;
}

.c12:focus-visible:enabled {
  background-color: #C7D6F7;
  outline: none;
}

.c14 {
  position: relative;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transform: translate3d(0,0,0.0625rem);
  -ms-transform: translate3d(0,0,0.0625rem);
  transform: translate3d(0,0,0.0625rem);
}

.c5 {
  position: relative;
  width: 100%;
}

.c10 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #808080;
  grid-area: label;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c11 {
  grid-area: right-icon;
}

.c8 {
  width: 100%;
  height: 1.5625rem;
  padding: 0 1rem;
  padding-right: 0.5rem;
  box-sizing: border-box;
  display: grid;
  grid-template-areas: ' label right-icon';
  grid-template-columns: 1fr 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
  background-color: #FFFFFF;
  border: 0.0625rem solid rgba(0,0,0,0.08);
  border-top-width: 0;
  -webkit-transition: all 300ms cubic-bezier(0.45,0.05,0,1);
  transition: all 300ms cubic-bezier(0.45,0.05,0,1);
}

.c8:hover {
  cursor: pointer;
}

.c9 {
  border-top-width: 0.0625rem;
  border-color: !important;
  border-radius: 0.25rem;
}

.c9:hover {
  border-color: #2F65E0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  line-height: 1.75rem;
}

.c1 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #4475E3;
  line-height: 1.75rem;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: -0.125rem;
}

.c13 {
  margin-left: 1rem;
}

.c13:disabled > div {
  background-color: transparent;
}

.c13:disabled > div svg {
  fill: #DBDBDB;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 2.5625rem;
}

.c4 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #4475E3;
  margin-right: 0.5rem;
}

.c6 {
  width: 2.5rem;
  height: 1.5625rem;
  border: none;
  border-radius: 0.25rem;
  z-index: 10;
  color: #808080;
}

.c6 > .c7 {
  position: relative;
  z-index: 100;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-left: 0.375rem;
  padding-right: 0.375rem;
  border-color: #DBDBDB;
}

.c6 > .c7:hover {
  cursor: pointer;
}

.c6 > .c7 > svg {
  display: none;
}

<body>
  <div>
    <div
      class="c0"
      data-testid="pagination"
    >
      <div
        class="c1"
        data-testid="info"
      >
        0--2 of -1
      </div>
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <span
            class="c4"
          >
            Rows per page
          </span>
          <div
            class="c5 c6"
          >
            <div
              class="c7 c8 c9"
              data-testid="value-item"
            >
              <div
                class="c10"
              />
              <svg
                class="c11"
              />
            </div>
          </div>
        </div>
        <button
          aria-label="Go to First"
          class="c12 c13"
          data-testid="go-to-first"
          fill="text"
          style="overflow: hidden; position: relative;"
        >
          <div
            class="c14"
            data-testid="content"
          >
            <svg />
          </div>
          <div
            class="c15"
            color="primaryLightPressed"
            opacity="1,1"
            scale="0"
            style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
          />
        </button>
        <button
          aria-label="Go to Previous"
          class="c12 c13"
          data-testid="go-to-previous"
          fill="text"
          style="overflow: hidden; position: relative;"
        >
          <div
            class="c14"
            data-testid="content"
          >
            <svg />
          </div>
          <div
            class="c15"
            color="primaryLightPressed"
            opacity="1,1"
            scale="0"
            style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
          />
        </button>
        <button
          aria-label="Go to Next"
          class="c12 c13"
          data-testid="go-to-next"
          fill="text"
          style="overflow: hidden; position: relative;"
        >
          <div
            class="c14"
            data-testid="content"
          >
            <svg />
          </div>
          <div
            class="c15"
            color="primaryLightPressed"
            opacity="1,1"
            scale="0"
            style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
          />
        </button>
        <button
          aria-label="Go to Last"
          class="c12 c13"
          data-testid="go-to-last"
          fill="text"
          style="overflow: hidden; position: relative;"
        >
          <div
            class="c14"
            data-testid="content"
          >
            <svg />
          </div>
          <div
            class="c15"
            color="primaryLightPressed"
            opacity="1,1"
            scale="0"
            style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`Pagination test pagination render 1`] = `
.c15 {
  position: absolute;
  will-change: opacity,transform;
  border-radius: 50%;
  -webkit-animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  z-index: 0;
  background-color: #E3EAFB;
  pointer-events: none;
}

.c12 {
  position: relative;
  z-index: 0;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 100%;
  white-space: nowrap;
  padding: 0;
  -webkit-transition-duration: 0;
  transition-duration: 0;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 1.5rem;
  width: 1.5rem;
  border: none;
  background-color: transparent;
}

.c12:hover {
  cursor: pointer;
}

.c12 > div {
  color: #4475E3;
}

.c12 > div svg {
  fill: #4475E3;
}

.c12:hover:enabled > div {
  color: #2F65E0;
}

.c12:hover:enabled > div svg {
  fill: #2F65E0;
}

.c12:hover:enabled:active > div {
  color: #2057D5;
}

.c12:hover:enabled:active > div svg {
  fill: #2057D5;
}

.c12:disabled > div {
  color: #B3C6F1;
}

.c12:disabled > div svg {
  fill: #B3C6F1;
}

.c12:focus-visible {
  outline: 0.125rem solid #C7D6F7;
}

.c12:focus-visible > div {
  color: #4475E3;
}

.c12:focus-visible > div svg {
  fill: #4475E3;
}

.c12:hover:enabled {
  background-color: #ECF1FC;
}

.c12:hover:enabled:active > div {
  color: #2057D5;
}

.c12:hover:enabled:active > div svg {
  fill: #2057D5;
}

.c12:focus-visible:enabled {
  background-color: #C7D6F7;
  outline: none;
}

.c14 {
  position: relative;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transform: translate3d(0,0,0.0625rem);
  -ms-transform: translate3d(0,0,0.0625rem);
  transform: translate3d(0,0,0.0625rem);
}

.c5 {
  position: relative;
  width: 100%;
}

.c10 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  grid-area: label;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c11 {
  grid-area: right-icon;
}

.c8 {
  width: 100%;
  height: 1.5625rem;
  padding: 0 1rem;
  padding-right: 0.5rem;
  box-sizing: border-box;
  display: grid;
  grid-template-areas: ' label right-icon';
  grid-template-columns: 1fr 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
  background-color: #FFFFFF;
  border: 0.0625rem solid rgba(0,0,0,0.08);
  border-top-width: 0;
  -webkit-transition: all 300ms cubic-bezier(0.45,0.05,0,1);
  transition: all 300ms cubic-bezier(0.45,0.05,0,1);
}

.c8:hover {
  cursor: pointer;
}

.c9 {
  border-top-width: 0.0625rem;
  border-color: !important;
  border-radius: 0.25rem;
}

.c9:hover {
  border-color: #2F65E0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  line-height: 1.75rem;
}

.c1 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #4475E3;
  line-height: 1.75rem;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: -0.125rem;
}

.c13 {
  margin-left: 1rem;
}

.c13:disabled > div {
  background-color: transparent;
}

.c13:disabled > div svg {
  fill: #DBDBDB;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 2.5625rem;
}

.c4 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #4475E3;
  margin-right: 0.5rem;
}

.c6 {
  width: 2.5rem;
  height: 1.5625rem;
  border: none;
  border-radius: 0.25rem;
  z-index: 10;
  color: #808080;
}

.c6 > .c7 {
  position: relative;
  z-index: 100;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-left: 0.375rem;
  padding-right: 0.375rem;
  border-color: #DBDBDB;
}

.c6 > .c7:hover {
  cursor: pointer;
}

.c6 > .c7 > svg {
  display: none;
}

<body>
  <div>
    <div
      class="c0"
      data-testid="pagination"
    >
      <div
        class="c1"
        data-testid="info"
      >
        41-50 of 1000
      </div>
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <span
            class="c4"
          >
            Rows per page
          </span>
          <div
            class="c5 c6"
          >
            <div
              class="c7 c8 c9"
              data-testid="value-item"
            >
              <div
                class="c10"
              >
                10
              </div>
              <svg
                class="c11"
              />
            </div>
          </div>
        </div>
        <button
          aria-label="Go to First"
          class="c12 c13"
          data-testid="go-to-first"
          fill="text"
          style="overflow: hidden; position: relative;"
        >
          <div
            class="c14"
            data-testid="content"
          >
            <svg />
          </div>
          <div
            class="c15"
            color="primaryLightPressed"
            opacity="1,1"
            scale="0"
            style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
          />
        </button>
        <button
          aria-label="Go to Previous"
          class="c12 c13"
          data-testid="go-to-previous"
          fill="text"
          style="overflow: hidden; position: relative;"
        >
          <div
            class="c14"
            data-testid="content"
          >
            <svg />
          </div>
          <div
            class="c15"
            color="primaryLightPressed"
            opacity="1,1"
            scale="0"
            style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
          />
        </button>
        <button
          aria-label="Go to Next"
          class="c12 c13"
          data-testid="go-to-next"
          fill="text"
          style="overflow: hidden; position: relative;"
        >
          <div
            class="c14"
            data-testid="content"
          >
            <svg />
          </div>
          <div
            class="c15"
            color="primaryLightPressed"
            opacity="1,1"
            scale="0"
            style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
          />
        </button>
        <button
          aria-label="Go to Last"
          class="c12 c13"
          data-testid="go-to-last"
          fill="text"
          style="overflow: hidden; position: relative;"
        >
          <div
            class="c14"
            data-testid="content"
          >
            <svg />
          </div>
          <div
            class="c15"
            color="primaryLightPressed"
            opacity="1,1"
            scale="0"
            style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;
